ddr/lib/ddr-ir/TypePrinter.cpp

```diff
@@ -0,0 +1,173 @@
+/*******************************************************************************
+ * Copyright (c) 2018, 2018 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at http://eclipse.org/legal/epl-2.0
+ * or the Apache License, Version 2.0 which accompanies this distribution
+ * and is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License, v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception [1] and GNU General Public
+ * License, version 2 with the OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#include "ddr/ir/TypePrinter.hpp"
+
+#include "ddr/ir/ClassUDT.hpp"
+#include "ddr/ir/EnumUDT.hpp"
+#include "ddr/ir/Field.hpp"
+#include "ddr/ir/TypedefUDT.hpp"
+#include "ddr/ir/UnionUDT.hpp"
+
+void
+TypePrinter::printIndent() const
+{
+	for (int32_t i = _indent; i > 0; --i) {
+		printf("  ");
```

Comment:
Only question is should we be using the PortLibrary tty_printf instead of printf in this class and ddrgen.cpp?

Reply:
Perhaps that would be better, but this is only a debugging aid.